crates/telemetry/src/state.rs

```diff
@@ -968,6 +968,38 @@ impl TelemetryState {
         self.backlog_size.observe(&cx, total, labels);
     }
 
+    /// Inserts in the backlog a new event for the given sequence number.
+    /// This happens when the relayer observed a new SendPacket event.
+    pub fn update_backlog(
+        &self,
+        sequences: Vec<u64>,
+        chain_id: &ChainId,
+        channel_id: &ChannelId,
+        port_id: &PortId,
+        counterparty_chain_id: &ChainId,
+    ) {
+        // Unique identifier for a chain/channel/port.
+        let path_uid: PathIdentifier = PathIdentifier::new(
+            chain_id.to_string(),
+            channel_id.to_string(),
+            port_id.to_string(),
+        );
+
+        // Remove any sequence number from the backlog which isn't in the list of queried pending packets
+        // as they might have been relayed without the Hermes instance observing it
+        if let Some(path_backlog) = self.backlogs.get(&path_uid) {
+            let backlog = path_backlog.value();
+            let keys_to_remove: Vec<u64> = backlog
+                .iter()
+                .filter(|entry| !sequences.contains(entry.key()))
+                .map(|entry| *entry.key())
+                .collect();
+            for key in keys_to_remove.iter() {
+                self.backlog_remove(*key, chain_id, channel_id, port_id, counterparty_chain_id)
```

Comment:
This looks all fine to me, but since we are basically updating the backlog to only contain the given sequence numbers, I wonder if it would be possible to simply override it altogether instead, eg. by clearing it and calling `backlog_insert` for each sequence number?

But perhaps it's not simpler than what we are doing, and since the code is totally fine as is, feel free to ignore.

Reply:
Looking at it more closely I see advantages and disadvantages in overriding the backlog completly.

Pros:
* If the instance missed a SendPacket the backlog will correctly display the number of pending packets
Cons:
* The timestamp when inserting an entry will be wrong for already observed entries, https://github.com/informalsystems/hermes/blob/8acaf5db9099aad13896011f2fbfe9075b8d338b/crates/telemetry/src/state.rs#L918

I think the best solution would be to override the backlog and update the `backlog_oldest_timestamp` to be `updated_backlog_timestamp` which would contain the timestamp of the last time an entry was inserted or removed.
What do you think about this solution?